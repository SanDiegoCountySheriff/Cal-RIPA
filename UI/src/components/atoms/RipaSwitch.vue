<template>
  <div class="ripa-switch" :style="{ maxWidth: maxWidth + 'px' }">
    <v-switch
      v-model="model"
      class="v-input--reverse v-input--expand"
      :disabled="disabled"
      :hint="hint"
      :persistent-hint="hint.length > 0"
      :rules="rules"
      :hide-details="rules.length === 0"
      validate-on-blur
    >
      <template #label>
        {{ label }}
      </template>
    </v-switch>
  </div>
</template>

<script>
export default {
  name: 'ripa-switch',

  computed: {
    model: {
      get() {
        return this.viewModel
      },

      set(newVal) {
        this.$emit('input', newVal)
      },
    },
  },

  props: {
    value: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    label: {
      type: String,
      default: 'no',
    },
    hint: {
      type: String,
      default: '',
    },
    maxWidth: {
      type: Number,
      default: 250,
    },
    rules: {
      type: Array,
      default: () => [],
    },
  },
}
</script>

<style lang="scss">
.v-input--reverse .v-input__slot {
  flex-direction: row-reverse;
  justify-content: flex-end;
  .v-application--is-ltr & {
    .v-input--selection-controls__input {
      margin-right: 0;
      margin-left: 8px;
    }
  }
  .v-application--is-rtl & {
    .v-input--selection-controls__input {
      margin-left: 0;
      margin-right: 8px;
    }
  }
}

.v-input--expand .v-input__slot {
  .v-label {
    display: block;
    flex: 1;
  }
}
</style>
