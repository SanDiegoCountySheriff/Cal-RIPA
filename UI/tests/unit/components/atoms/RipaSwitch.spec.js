import RipaSwitch from '@/components/atoms/RipaSwitch'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Switch', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {})
  })

  const factory = propsData => {
    return shallowMount(RipaSwitch, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaSwitch, { vuetify })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should set model', async () => {
    wrapper = factory()

    wrapper.vm.model = 'New Value'
    await wrapper.vm.$nextTick()

    expect(wrapper.emitted('input')).toBeTruthy()
  })
})
