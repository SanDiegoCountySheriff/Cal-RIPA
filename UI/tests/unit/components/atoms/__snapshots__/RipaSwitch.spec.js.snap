// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Switch should match snapshot 1`] = `
<div class="ripa-switch" style="max-width: 250px;">
  <v-switch-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" validateonblur="true" backgroundcolor="" hidedetails="true" hint="" ripple="true" valuecomparator="[Function]" class="v-input--reverse v-input--expand"></v-switch-stub>
</div>
`;
